spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourcePoolFilterSpecTest.java

```diff
@@ -0,0 +1,254 @@
+        assertEquals(2, pools.size());
+    }
+
+    @Test
+    public void shouldAllowOwnersFilterPoolsByActivationKey() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ActivationKeyDTO activationKey = adminClient.owners().createActivationKey(
+            owner.getKey(), ActivationKeys.random(owner));
+
+        ProductDTO product1 = createProduct(adminClient, owner,
+            ProductAttributes.SupportLevel.withValue("VIP")
+        );
+        ProductDTO product2 = createProduct(adminClient, owner,
+            ProductAttributes.SupportLevel.withValue("Supurb"),
+            ProductAttributes.Cores.withValue("4")
+        );
+        ProductDTO product3 = createProductByOwner(adminClient, owner, Products.random());
+
+        PoolDTO pool1 = adminClient.owners().createPool(owner.getKey(), Pools.random(product1));
+        PoolDTO pool2 = adminClient.owners().createPool(owner.getKey(), Pools.random(product2));
+        adminClient.owners().createPool(owner.getKey(), Pools.random(product3));
+
+        adminClient.activationKeys().addPoolToKey(activationKey.getId(), pool1.getId(), null);
+        adminClient.activationKeys().addPoolToKey(activationKey.getId(), pool2.getId(), null);
+
+        // ignores the pools already with the activation key
+        List<PoolDTO> pools = adminClient.owners().listOwnerPools(owner.getKey(), null,
+            activationKey.getName(), null, null, null, null, null, null, null, null,
+            null, null, null, null, null, null);
+        assertEquals(1, pools.size());
+    }
+
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    public class PoolAndSubscriptionFiltering {
+
+        ApiClient adminClient;
+        private OwnerDTO owner;
+        private ProductDTO product1;
+        private ProductDTO product2;
+
+        @BeforeAll
+        public void setUp() throws ApiException {
+            adminClient = ApiClients.admin();
+            owner = createOwner(adminClient);
+
+            product1 = createProduct(adminClient, owner,
+                ProductAttributes.SupportLevel.withValue("VIP")
+            );
+            product2 = createProduct(adminClient, owner,
+                ProductAttributes.SupportLevel.withValue("Supurb"),
+                ProductAttributes.Cores.withValue("4")
+            );
+            ProductDTO product3 = createProductByOwner(adminClient, owner, Products.random());
+
+            createPools(adminClient, owner, Pools.random(product1),
+                Pools.random(product2), Pools.random(product3));
+        }
+
+        @Test
+        public void shouldAllowOwnersFilterPoolsBySingleFilter() throws ApiException {
+            List<PoolDTO> filteredPools = adminClient.owners().listOwnerPoolsWithAttributes(
+                owner.getKey(), List.of("support_level:VIP"));
+            assertEquals(1, filteredPools.size());
```

Comment:
These could be combined into one assertion chain that includes the missing null checks:

```java
assertThat(filteredPools)
  .isNotNull()
  .singleElement()
  .isNotNull()
  .returns(product1.getId(), PoolDTO::getProductId);
```

This could be applied in several places.

Reply:
done, I replaced it in other files where it seemed appropriate, so I hope I didn't forget it somewhere